pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TopicTransactionBufferRecoverTest.java

```diff
@@ -502,6 +502,32 @@ public void testTransactionBufferRecoverThrowBrokerMetadataException() throws Ex
 
         producer.newMessage(txn).value("test".getBytes()).sendAsync();
         txn.commit().get();
+
+        PersistentTopic newTopic = (PersistentTopic) getPulsarServiceList().get(0)
+                .getBrokerService().getTopic(TopicName.get(topic).toString(), false).get().get();
+        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+        completableFuture.completeExceptionally(new PulsarClientException.BrokerPersistenceException("test"));
+        doReturn(completableFuture)
+                .when(transactionBufferSnapshotService).createReader(any());
```

Comment:
Interesting, why create reader will get BrokerPersistenceException, it will not publish messages to the topic.